Source/missiles.cpp

```diff
@@ -4391,125 +4388,80 @@ void __fastcall MI_Firewall(int i)
 		}
 	} else {
-		missile[v1]._mitxoff += missile[v1]._mixvel;
-		missile[v1]._mityoff += missile[v1]._miyvel;
-		GetMissilePos(ia);
-		v11 = missile[v1]._mix;
-		if (v11 != missile[v1]._misx || missile[v1]._miy != missile[v1]._misy)
-			CheckMissileCol(ia, v5, v5, 0, v11, missile[v1]._miy, 0);
-		v12 = missile[v1]._mix;
-		if (missile[v1]._mirange) {
-			if (v12 != missile[v1]._miVar1 || missile[v1]._miy != missile[v1]._miVar2) {
-				v14 = missile[v1]._mlid;
-				missile[v1]._miVar1 = v12;
-				v15 = missile[v1]._miy;
-				missile[v1]._miVar2 = v15;
-				ChangeLight(v14, v12, v15, 8);
-			}
-		} else {
-			tx = missile[v1]._mix;
-			ty = missile[v1]._miy;
-			ChangeLight(missile[v1]._mlid, v12, ty, missile[v1]._miAnimFrame);
-			if (!CheckBlock(fx, fy, tx, ty))
-				CheckMissileCol(ia, v5, v5, 0, tx, ty, 1);
-			if (!CheckBlock(fx, fy, tx, ty + 1))
-				CheckMissileCol(ia, v5, v5, 0, tx, ty + 1, 1);
-			if (!CheckBlock(fx, fy, tx, ty - 1))
-				CheckMissileCol(ia, v5, v5, 0, tx, ty - 1, 1);
-			if (!CheckBlock(fx, fy, tx + 1, ty))
-				CheckMissileCol(ia, v5, v5, 0, tx + 1, ty, 1);
-			if (!CheckBlock(fx, fy, tx + 1, ty - 1))
-				CheckMissileCol(ia, v5, v5, 0, tx + 1, ty - 1, 1); /* check x/y */
-			if (!CheckBlock(fx, fy, tx + 1, ty + 1))
-				CheckMissileCol(ia, v5, v5, 0, tx + 1, ty + 1, 1);
-			if (!CheckBlock(fx, fy, tx - 1, ty))
-				CheckMissileCol(ia, v5, v5, 0, tx - 1, ty, 1);
-			if (!CheckBlock(fx, fy, tx - 1, ty + 1))
-				CheckMissileCol(ia, v5, v5, 0, tx - 1, ty + 1, 1);
-			if (!CheckBlock(fx, fy, tx - 1, ty - 1))
-				CheckMissileCol(ia, v5, v5, 0, tx - 1, ty - 1, 1);
-			v13 = 112 * tx + ty;
-			if (!TransList[dung_map[0][v13]] /* check */
-			    || missile[v1]._mixvel < 0
-			        && (TransList[dung_map[0][v13 + 1]] && nSolidTable[dPiece[0][v13 + 1]]
-			               || TransList[dung_map[0][v13 - 1]] && nSolidTable[dPiece[0][v13 - 1]])) // TransList[*(&byte_5B78EB + v13)] && nSolidTable[*(_DWORD *)&dflags[39][4 * v13 + 36]]) )
-			{
-				++missile[v1]._mix;
-				++missile[v1]._miy;
-				missile[v1]._miyoff -= 32;
+		missile[i]._mitxoff += missile[i]._mixvel;
+		missile[i]._mityoff += missile[i]._miyvel;
+		GetMissilePos(i);
+		if (missile[i]._mix != missile[i]._misx || missile[i]._miy != missile[i]._misy)
+			CheckMissileCol(i, dam, dam, 0, missile[i]._mix, missile[i]._miy, 0);
+		if (!missile[i]._mirange) {
+			mx = missile[i]._mix;
+			my = missile[i]._miy;
+			ChangeLight(missile[i]._mlid, missile[i]._mix, my, missile[i]._miAnimFrame);
+			if (!CheckBlock(px, py, mx, my))
+				CheckMissileCol(i, dam, dam, 0, mx, my, 1);
+			if (!CheckBlock(px, py, mx, my + 1))
+				CheckMissileCol(i, dam, dam, 0, mx, my + 1, 1);
+			if (!CheckBlock(px, py, mx, my - 1))
+				CheckMissileCol(i, dam, dam, 0, mx, my - 1, 1);
+			if (!CheckBlock(px, py, mx + 1, my))
+				CheckMissileCol(i, dam, dam, 0, mx + 1, my, 1);
+			if (!CheckBlock(px, py, mx + 1, my - 1))
+				CheckMissileCol(i, dam, dam, 0, mx + 1, my - 1, 1); /* check x/y */

```

Comment:
What is the x/y comment about?

Reply:
I think @galaxyhaxz wanted to check that the values where correct, I didn't remove it since the memory is still off and so we can't be 100% sure it's correct even though the operations are. But I think it's fine to remove it since if there is an issue it will eventually get caught when we start cleaning up the memory layout.

Reply:
I'm fine either way. If @galaxyhaxz want's the comment, then let it stay for now. Any chance to bring her back to the party is a chance I welcome wholeheartedly :heart: